### ISSUE TITLE
Case sensitive usernames

### ISSUE THREAD

**Issue by JMBokhorst** [nextcloud/server#10814]
### Steps to reproduce
1. Create user with Capital
2. Login without Capital on Windows desktop app (not tested on any other OS) 
3. Client will login, and let's you select folders. However after finishing the setup, no user-data/folders can be found. 

### Expected behaviour
Or the client should through an error that the username doesn't exists. Or the client should not be case-sensitive. 

### Actual behaviour
Client fails to retrive folder information or sync. 

### Server configuration

**Operating system**:
Ubuntu 16.04

**Web server:**
Apache v2.4.29
**Database:**
mysql 
**PHP version:**
7.2.7
**Nextcloud version:** (see Nextcloud admin page)
13.0.5.2
**Updated from an older Nextcloud/ownCloud or fresh install:**
Updated.
**Where did you install Nextcloud from:**
Via the nextcloud server manual on the website.

**Signing status:**
<details>
<summary>Signing status</summary>

No errors have been found.</details>

**List of activated apps:**
<details>
<summary>App list</summary>

Enabled:
  - activity: 2.6.1
  - admin_audit: 1.3.0
  - bruteforcesettings: 1.1.0
  - calendar: 1.6.1
  - checksum: 0.4.0
  - comments: 1.3.0
  - contacts: 2.1.5
  - dav: 1.4.7
  - drawio: 0.9.0
  - encryption: 2.0.0
  - federatedfilesharing: 1.3.1
  - federation: 1.3.0
  - files: 1.8.0
  - files_accesscontrol: 1.3.0
  - files_downloadactivity: 1.2.0
  - files_pdfviewer: 1.2.1
  - files_reader: 1.2.3
  - files_sharing: 1.5.0
  - files_texteditor: 2.5.1
  - files_trashbin: 1.3.0
  - files_versions: 1.6.0
  - files_videoplayer: 1.2.0
  - firstrunwizard: 2.2.1
  - flowupload: 0.0.7
  - gallery: 18.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.1.0
  - nextcloud_announcements: 1.2.0
  - notes: 2.4.0
  - notifications: 2.1.2
  - oauth2: 1.1.1
  - onlyoffice: 2.0.2
  - passman: 2.1.4
  - password_policy: 1.3.0
  - provisioning_api: 1.3.0
  - quota_warning: 1.2.0
  - ransomware_protection: 1.1.1
  - serverinfo: 1.3.0
  - sharebymail: 1.3.0
  - survey_client: 1.1.0
  - systemtags: 1.3.0
  - tasks: 0.9.6
  - theming: 1.4.5
  - twofactor_backupcodes: 1.2.3
  - updatenotification: 1.3.0
  - workflowengine: 1.3.0
Disabled:
  - files_external
  - richdocuments
  - user_external
  - user_ldap
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "nextcloud.bokhorst.info",
            "owncloud.bokhorst.info"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "http:\/\/nextcloud.bokhorst.info",
        "dbtype": "mysql",
        "version": "13.0.5.2",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "filelocking.enabled": "true",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "timeout": 0
        },
        "mail_smtpmode": "php",
        "mail_smtpauthtype": "LOGIN",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "theme": "",
        "log_type": "owncloud",
        "logfile": "\/var\/log\/nextcloud.log",
        "loglevel": "0",
        "log_rotate_size": "104857600",
        "updater.release.channel": "stable",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...
No

**Are you using encryption:** yes/no
Yes 

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
No 

### Client configuration
**Browser:**
None, Windows client (latest version) 

**Operating system:**
Windows 10 64-bit

**Comment by blizzz** [comment]
@rullzer were there changes in the client in that regard?

**Comment by rullzer** [comment]
nah I think this is the username vs userid issue. since it uses the new endpoint. I would need to check

**Comment by Niklas974** [comment]
Additional Issue this bug causes:
Using QOwnNotes client software, which accesses QOwnNotesAPI which is a Plugin only works with the non-capitalized Username. (I can login the webinterface with either capitalized or non-capitalized version of the username.)
This is a Problem as the Calendar seems to use the capitalized Username. It is unintuitive for the client software to query for a calendar name using a capitalized username when login (only) works with a non-capitalized username.

tl;dr: Loggin in against a plugin provided API does not allow using capitalized Usernames.
See also the issue I mistakenly opened at [QOwnNotes](https://github.com/pbek/QOwnNotes/issues/1134)

**Comment by skjnldsv** [comment]
@rullzer we fixed it, right?
